src/Providers/cordova-ironsource.ts

```diff
@@ -0,0 +1,77 @@
+declare var IronSourceAds: any | undefined
+
+import { IProvider } from './ad-provider'
+import { AdType, AdEvents, H5AdWrapper } from '../h5-ad-wrapper'
+
+export class CordovaIronSource implements IProvider {
+    public adManager!: H5AdWrapper
+    public adsEnabled: boolean = false
+
+    private interstitialLoaded: boolean = false
+
+    constructor(appKey: string) {
+        if (typeof IronSourceAds === 'undefined') {
+            return
+        }
+
+        IronSourceAds.init({
+            appKey: appKey
+        })
+        window.addEventListener('interstitialClosed', () => {
+            this.resumeGameplay()
+        })
+        window.addEventListener('interstitialLoaded', () => {
+            this.interstitialLoaded = true
+        })
+        window.addEventListener('rewardedVideoClosed', () => {
+            this.resumeGameplay()
+        })
+    }
+
+    public setManager(manager: H5AdWrapper): void {
+        this.adManager = manager
+    }
+
+    public showAd(adType: AdType = AdType.interstitial): void {
+        switch (adType) {
+            case AdType.interstitial:
+                if (!this.interstitialLoaded) {
+                    this.resumeGameplay()
+                    break
+                }
+                this.interstitialLoaded = false
+                this.adManager.emit(AdEvents.CONTENT_PAUSED)
+                IronSourceAds.showInterstitial()
+                break
+            case AdType.rewarded:
+                this.adManager.emit(AdEvents.CONTENT_PAUSED)
+                IronSourceAds.showRewardedVideo()
+                break
+            default:
+                this.resumeGameplay()
+                break
+        }
+    }
+
+    private resumeGameplay(): void {
+        this.adManager.emit(AdEvents.CONTENT_RESUMED)
+    }
+
+    public preloadAd(adType: AdType = AdType.interstitial): void {
+        if (adType === AdType.interstitial) {
+            IronSourceAds.loadInterstitial()
+        }
+    }
+
+    public destroyAd(): void {
+        return
+    }
+
+    public hideAd(): void {
+        return
+    }
+
+    public adAvailable(adType: AdType): boolean {
+        return true
```

Comment:
So if AdType.interstitial, then you can return this.interstitialLoaded? Or sRewardedVideoAvailable()?
Otherwise if AdType.rewarded, then return isInterstitialReady()?